### PR TITLE
fixes for MQX classic 4.0 with IAR-EWARM

### PR DESCRIPTION
**Pull request by cconlon**
This PR includes fixes for building wolfSSL for MQX 4.0 (an older version) using IAR-EWARM.

This adds a new define called **FREESCALE_MQX_4_0** that automatically defines **FREESCALE_MQX** but also makes a few minor adjustments to the build.

IAR-EWARM threw warnings for the code sections that used the "~" operator, similar to:
```
internal.c
Remark[PA091]: operator operates on value promoted to int (with possible unexpected result)
```
The casts to (word32) were to address these warnings.

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -12062,7 +12062,7 @@ int TimingPadVerify(WOLFSSL* ssl, const byte* input, int padLen, int macSz,
     good |= MaskMac(input, pLen, ssl->specs.hash_size, verify);
 
     /* Non-zero on failure. */
-    good = ~good;
+    good = ~(word32)good;
```
Comment (toddouska) [inline src/internal.c]: These are the same type, why is this needed?

Comment (toddouska) [reply src/internal.c]: Can you run these casts by Sean?

Comment (cconlon) [reply src/internal.c]: Sure, I will run these by Sean.

Based on my research, the "~" operator first promotes "good" from a byte (unsigned char) to a signed integer here before doing the operation.  Since semantics for a signed int can vary, casting to an unsigned integer guarantees the value used is the unsigned binary as expected.

Comment (toddouska) [reply src/internal.c]: Are you positive that none of our supported compilers will complain about assigning 32 bits to 8 bits?

Comment (cconlon) [reply src/internal.c]: Not completely, just for the compilers I have tested so far:

- Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn) - OSX 10.9.5
- gcc 7.2.0 - Ubuntu 17.10
- clang 4.0.1-6 - Ubuntu 17.10
- Visual Studio 2015 Community - Windows 10
- IAR-EWARM 6.40.5.4275 - Windows 7 Pro

One option would be to cast the result back to the original type, which should suppress any 32 to 8-bit or similar warnings:

```
good = (byte)~(word32)good;
```

The other option would be to suppress the [Pa091] warning when building for IAR-EWARM projects.
